### DISCUSSION TITLE
Proposal for visualization of difficulty change from applying mods

### DISCUSSION THREAD

**Discussion by zachllogan** [ppy/osu#12995] [Ideas]
Sometimes when I'm playing osu with mods, I'm curious as to what star level the map was before applying the mod, but I don't want to switch to no mod and back to find out. So I created this:
![modincreaseddifficultystarvisualization](https://user-images.githubusercontent.com/22415468/44435389-663f9600-a575-11e8-8d3e-2277a0d7710d.PNG)
![moddecreaseddifficultystarvisualization](https://user-images.githubusercontent.com/22415468/44435391-68095980-a575-11e8-9a73-27676a32b825.PNG)
The original star level is preserved in white and extra stars in yellow or green (matching the color for the type in the mod select menu) are added to show the star level after applying all mods.
I'm not sure this is any good, so if you don't like it feel free to say so and close the issue, but if you do like it, I have code I can PR.

Note: For this I also created code to dynamically update star difficulty to account for selected mods & modes. It works by creating a difficulty calculator from the ruleset and then using that to calculate updated difficulties. Do you want me to PR this? I know you guys have been working on database stuff to do that anyways, so you might not want this implementation.

**Comment by smoogipoo** [comment]
This would be pretty cool to see.

**Comment by zachllogan** [comment]
@smoogipoo Ok. Then, if I PR it, should I also PR my code for dynamically updating star difficulty, and if I do that, should I PR them separately or in one merged PR?

**Comment by smoogipoo** [comment]
Separate is good.
